File: Lab 5/TFigure.py
from abc import ABC, abstractmethod
class TFigure(ABC):
    def __init__(self, height: float):
        self._height = height
    
    @property
    def height(self):
        return self._height

    @height.setter
    def height(self, height):
        self._height 

    @abstractmethod
    def GetS(self):
        pass
    
    @abstractmethod
    def GetV(self):
        pass
    
    @abstractmethod
    def GetSOs(self):
        pass

File: Lab 5/TFigure.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab_5
{
    internal abstract class TFigure
    {
        protected double height;
        public double Height { get { return height; } set { height = value; } }

        protected TFigure(double height)
        {
            this.height = height;
        }
        abstract public double GetS();
        abstract public double GetV();
        abstract protected double GetSOS();
    }
}
